### ISSUE TITLE
Implement *~ operation

### ISSUE THREAD

**Issue by drodiger** [mesarth/JSONPath-Notebook#14]
$.* usually returns all objects
$.*~ should return only first children key names (not values).
$.data.*~ would return all children key names under data.
$.*.*~ would return all second level key names.

jsonpath.com has this implemented, you can test.

With this operation it is easier to analyse data in json.

**Comment by mesarth** [comment]
This syntax is not part of the [JSONPath standard](https://www.rfc-editor.org/rfc/rfc9535.html). I don't think it's a good idea to add additional operations on top of the standard. 

Maybe we can add a toggle button to the notebook cell toolbar to switch between values and key names.

**Comment by drodiger** [comment]
That would also help. Or switch between standard JSON and non standard ;-)

**Comment by jg-rp** [comment]
Version 1.2.0 of JSON P3 includes a non-standard _keys selector_ that might be of interest to you ([docs](https://jg-rp.github.io/json-p3/guides/jsonpath-extra)).

It is disabled by default. Set the `strict` option to `false` when constructing a `JSONPathEnvironment` to enable all non-standard features.

When enabled, the keys selector defaults to `~`. You can change it to `*~` by setting the `keysPattern` option to `/\*~/y`.

```javascript
import { JSONPathEnvironment } from "json-p3";

const standardEnv = new JSONPathEnvironment();
const nonStandardEnv = new JSONPathEnvironment({ strict: false, keysPattern: /\*~/y });
const data = {data: {hello: "foo", world: "bar"}};

console.log(nonStandardEnv.query("$.data.*~", data).values());
// [ 'hello', 'world' ]

standardEnv.query("$.data.*~", data);
// JSONPathSyntaxError: expected '.', '..' or a bracketed selection, found '~' ('$.data.*~':8)
```

**Comment by mesarth** [comment]
Thank you @jg-rp! That's great!

I've added an icon button in the cell's status bar to switch between standard and non-standard syntax modes.
<img width="829" alt="scrennshot" src="https://github.com/mesarth/JSONPath-Notebook/assets/6097749/aaa3d42c-b631-44fb-ad72-c00c0d66246e">

The default mode can be changed in the settings.
![image](https://github.com/mesarth/JSONPath-Notebook/assets/6097749/93fdd321-f966-400b-8c14-320b499cdcb7)

Does this solve your problem @drodiger?

**Comment by drodiger** [comment]
Yes, that would be great. Thanks

**Comment by mesarth** [comment]
@drodiger this is now available with version 2.2.0

**Comment by drodiger** [comment]
Tested 2.2.0. Works great. You can close the request. Thanks
